controllers/storagecluster/storageclasses.go

```diff
@@ -103,6 +104,27 @@ func (r *StorageClusterReconciler) createStorageClasses(sccs []StorageClassConfi
 				skippedSC = append(skippedSC, sc.Name)
 				continue
 			}
+		case strings.Contains(sc.Name, "-ceph-replica1-rbd") && !scc.isClusterExternal:
+			// wait for CephBlockPools to be ready
+			cephBlockPools := cephv1.CephBlockPoolList{}
+			err := r.Client.List(context.TODO(), &cephBlockPools, client.InNamespace(sc.Parameters["clusterID"]))
+			num := len(initData.Status.FailureDomainValues)
```

Comment:
Rather than passing in the StorageCluster spec at all, just read the information from the StorageClass parameters:

```suggestion
		num := len(scc.StorageClass.Parameters["topologyConstrainedPools"])
```

You probably need to do `strings.Split()` or something.